### DISCUSSION TITLE
How can we send a shutdown HTTP request when the browser is closing?

### DISCUSSION THREAD

**Discussion by liebichw** [cefsharp/CefSharp#4025] [General]
A webapplication we have uses the JavaScript "unload" event to send a kind of "shutdown notification" HTTP request to a backend server.
Now this works, when the web application runs in a standalone Chrome browser.
When running it in a CefSharp-Browser (Version 92.0.26.0), the request does not arrive at the server, however.

I experimented with a testing webpage and discovered that if I block the "unload" event handler for at least 6ms, the HTTP request does arrive at the back end. A smaller delay (or no delay at all) does not work - the request does not make it to the server.

Is there any setting or method for CefSharp to make this "last HTTP request" work reliably without delaying the event handler by force? We cannot directly change the code of the web application in question, so the adaptation would have to be made via CefSharp. Is it possible?

**Comment by amaitland** [comment]
You can try closing the underlying `CefBrowser` manually

https://cefsharp.github.io/api/98.1.x/html/M_CefSharp_IBrowser_CloseBrowser.htm

be something like `chromiumWebBrowser.GetBrowser().CloseBrowser(false);`.

Currently `Dispose` calls `CloseBrowser(true)` as there's no real option to cancel out once you start disposing. 

The upstream docs have some of the full details https://github.com/chromiumembedded/cef/blob/ba46b8c53ed0f46cec82a1aebaec173784deb6e4/include/cef_life_span_handler.h#L107
